File: Server/src/main/java/com/motivastudy/demo/models/Disciplina.java
package com.motivastudy.demo.models;

import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import lombok.Data;

@Entity
@Data
@Table(name = "disciplina")
public class Disciplina {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;

    private String nome;

    @OneToMany(mappedBy = "disciplina")
    private List<Topico> topicos;
}

File: Client/motivastudy-client/src/app/models/Disciplina.ts
import { Topico } from "./Topico";

export class Disciplina {
  id:string;
  nome:string;
  topicos:Topico[]
  constructor(id:string, nome:string, topicos:Topico[]){
    this.id = id;
    this.nome = nome;
    this.topicos = topicos;
  }
}
